pkg/graph/resolvers/mto_milestone_test.go

```diff
@@ -0,0 +1,165 @@
+	if suite.NotNil(milestone1A.MTOCategoryID) {
+		suite.Equal(category1SubA.ID, *milestone1A.MTOCategoryID)
+	}
+	suite.Equal(suite.testConfigs.Principal.UserAccount.ID, milestone1A.CreatedBy)
+	suite.Nil(milestone1A.ModifiedBy)
+	suite.Nil(milestone1A.ModifiedDts)
+}
+
+func (suite *ResolverSuite) TestMTOMilestoneCreateCustomDuplicates() {
+	// Create model plan
+	plan := suite.createModelPlan("testing duplicate custom milestone creation")
+
+	// Create an uncategorized custom milestone
+	milestone, err := MTOMilestoneCreateCustom(suite.testConfigs.Context, suite.testConfigs.Logger, suite.testConfigs.Principal, suite.testConfigs.Store, "Custom Milestone", plan.ID, nil)
+	suite.NoError(err)
+	suite.NotNil(milestone)
+
+	// Try and create one with the same name
+	milestoneDupe, err := MTOMilestoneCreateCustom(suite.testConfigs.Context, suite.testConfigs.Logger, suite.testConfigs.Principal, suite.testConfigs.Store, "Custom Milestone", plan.ID, nil)
+	suite.Error(err)
+	suite.Nil(milestoneDupe)
+
+	// Try and create one with the same even though it's in a different category
+	category, err := MTOCategoryCreate(suite.testConfigs.Context, suite.testConfigs.Logger, suite.testConfigs.Principal, suite.testConfigs.Store, "Custom Category", plan.ID, nil)
+	suite.NoError(err)
+	milestoneDupe, err = MTOMilestoneCreateCustom(suite.testConfigs.Context, suite.testConfigs.Logger, suite.testConfigs.Principal, suite.testConfigs.Store, "Custom Milestone", plan.ID, &category.ID)
+	suite.Error(err)
+	suite.Nil(milestoneDupe)
+}
+
+func (suite *ResolverSuite) TestMTOMilestoneCreateCommon() {
+	// Create model plan
+	plan := suite.createModelPlan("testing common milestone creation")
+
+	// First, make sure there's no categories to start (except uncategorized)
+	categories, err := MTOCategoryGetByModelPlanIDLOADER(suite.testConfigs.Context, plan.ID)
+	suite.NoError(err)
+	suite.Len(categories, 1) // Just uncategorized
+
+	// Then, create a common milestone
+	milestone, err := MTOMilestoneCreateCommon(suite.testConfigs.Context, suite.testConfigs.Logger, suite.testConfigs.Principal, suite.testConfigs.Store, plan.ID, models.MTOCommonMilestoneKeyManageCd)
+	suite.NoError(err)
+	suite.Equal(suite.testConfigs.Principal.UserAccount.ID, milestone.CreatedBy)
+	suite.Nil(milestone.ModifiedBy)
+	suite.Nil(milestone.ModifiedDts)
+
+	// Check and see if the categories came along as expected
+	// You can check which we expect to get created with migrations/V190__Add_Common_Milestone_Library.sql (or future similar migrations that edit the library)
+	// In this case MTOCommonMilestoneKeyManageCd should create 2 categories (parent, subcategory) of ('Operations','Participant and beneficiary tracking')
+	categories, err = MTOCategoryGetByModelPlanIDLOADER(suite.testConfigs.Context, plan.ID)
+	suite.NoError(err)
+	suite.Len(categories, 2) // 1 new category (Operations) + Uncategorized
+
+	operationsCategory, found := lo.Find(categories, func(item *models.MTOCategory) bool {
```

Comment:
Now that the resolver is merged in, I might suggest calling `MTOCategoriesGetByID` to verify the category and subcategory. I do like that this is validating the category creation though!

Also fine to keep this the way it is since you are validating all categories for the MTO!

Reply:
I think I'm just going to add a comment for a potential change here -- I had a bit of trouble implementing `MTOCategoriesGetByID ` because I still ended up wanting to fetch the total list of categories/subcategories to ensure the proper number of categories existed post-creation (which `MTOCategoriesGetByID` doesn't necessarily validate)

Reply:
077d6784d